guids-generator.html

```diff
@@ -462,21 +462,23 @@
                             return;
                         }
 
-                        // Remove salt version, rounds number and actual salt from hash
-                        // Encode the hash in HEX to ensure only alphanumeric characters are generated
-                        hash = hash.replace(salt, '')
-                                 .split("")
-                                 .map((c, i) => {
-                                                   let str = c.charCodeAt(0).toString(16).padStart(2, "0");
-                                                   // Add random noise in alternating bytes
-                                                   str += Math.random().toString(16).substring(2, 4);
-                                                   return str;
-                                                })
-                                 .join("");
+                        /*
+                         Turn the base64 encoded hash (remove salt version, rounds number and actual salt)
+                         into a Typed Array that we can work with byte-by-byte.
+                        */
+                        hash = Uint8Array.from(window.atob(hash.slice(-31)), (c) => c.charCodeAt(0));
```

Comment:
Can the magic `-31` be explained in a comment? Or even better, can it be computed from other info about the algorithm, which would make it more resilient in case we want to change the hash method/length?

Reply:
This issue is resolved in the `phiHashFunction` introduced in the PR https://github.com/data-team-uhn/guid-generator-web/pull/19.